### ISSUE TITLE
standalone web service

### ISSUE THREAD

**Issue by npdoty** [npdoty/ephemerurl#1]
For people who can't easily install extra code on their domain, we should set up a standalone service at its own domain that will create ephemerurls for any URL on the Web. (h/t @sbenthall) 

While the primary use case is not to allow people to make ephemeral links to content they don't have any control over, that doesn't seem like a serious abuse, and it would have the benefit of letting people make ephemeral links to content they authored but where they don't have the access to or don't to install code on the server itself.

**Comment by sbenthall** [comment]
Expanding on the use case, there are pastebins with 'secret' but permanent URLs that you could share ephemerally with your service.

https://gist.github.com/

**Comment by npdoty** [comment]
it's like [perma.cc](https://perma.cc), except the opposite

**Comment by elsehow** [comment]
This could be done pretty simply with a persistent kv store. Keys represent the shortened URLs, and values represent the real URLs. Then we just need some service that deletes things from the kv store at the right times.

As an added bonus, it wouldn't require any client-side JS.

**Comment by npdoty** [comment]
Yeah, for the current self-hosted version, you just set up a sqlite database with the url, the random short string and an expiration. Client-side JavaScript is just an optional value at the target URL which reminds the user to please share only the ephemeral URL.

@elsehow, did you have a particular key-value store in mind that's better than sqlite-on-a-server-somewhere?

**Comment by elsehow** [comment]
I use leveldb and cousins, but I'm sure any KV store will do.
